src/rust/rust-proto-clients/src/grpc_client.rs

```diff
@@ -0,0 +1,66 @@
+use std::time::Duration;
+
+use rust_proto::protocol::{
+    endpoint::Endpoint,
+    healthcheck::client::HealthcheckClient,
+    service_client::{
+        ConnectError,
+        Connectable,
+        NamedService,
+    },
+};
+
+use crate::grpc_client_config::{
+    GenericGrpcClientConfig,
+    GrpcClientConfig,
+};
+
+pub struct GetGrpcClientOptions {
+    pub perform_healthcheck: bool,
+    pub healthcheck_polling_interval: Duration,
+}
+
+impl Default for GetGrpcClientOptions {
+    fn default() -> Self {
+        GetGrpcClientOptions {
+            perform_healthcheck: false,
+            healthcheck_polling_interval: Duration::from_millis(500),
+        }
+    }
+}
+
+pub async fn get_grpc_client<C: GrpcClientConfig>(
```

Comment:
can we instead call these something like `create_*` or `build_*`? They don't actually really "get" anything, instead they build a new thing every time you call one.

That said, wouldn't this make more sense as a _static method on the actual client_ than as a disembodied constructor in a different crate?